### PR TITLE
Updated maven surefire plugin to helidon-standalone-quickstart-mp

### PR DESCRIPTION
**Pull request by ksurendra**
Following updates were done to "helidon-standalone-quickstart-mp" example.

1. Updated helidon version to "2.0.0-M3"
2. Updated **maven-surefire-plugin** version from `<version.plugin.surefire>2.19.1</version.plugin.surefire>` to `<version.plugin.surefire>3.0.0-M3</version.plugin.surefire>`
3. Updated **maven-failsafe-plugin** version from `<version.plugin.failsafe>2.19.1</version.plugin.failsafe>` to `<version.plugin.failsafe>3.0.0-M3</version.plugin.failsafe>`
4. Removed `<version.plugin.surefire.provider.junit>1.0.3</version.plugin.surefire.provider.junit>`
5. Removed the below from plugin dependencies of **maven-surefire-plugin** and **maven-failsafe-plugin**
```
 <dependency>
   <groupId>org.junit.platform</groupId>
   <artifactId>junit-platform-surefire-provider</artifactId>
   <version>${version.plugin.surefire.provider.junit}</version>
 </dependency>
```

**Why the update?**
The version **3.0.0-M3** of **maven-surefire-plugin** supports Junit 5 natively and there is no need for **junit-platform-surefire-provider**. One less dependency for the project.

**Reference**: 
- https://github.com/junit-team/junit5/issues/1367#issuecomment-401055265

### CODE REVIEW

File: examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml [hunk 1]
```diff
@@ -22,11 +22,11 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>io.helidon.examples.quickstarts</groupId>
     <artifactId>helidon-standalone-quickstart-mp</artifactId>
-    <version>2.0.0-SNAPSHOT</version>
+    <version>2.0.0-M3</version>
```
Comment (tomas-langer) [inline examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml:25]: The change of version is not correct (for Helidon itself) - we handle that in our release process, to reference the currently released version.

File: examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml [hunk 2]
```diff
@@ -22,11 +22,11 @@
...
     <name>Helidon Standalone Quickstart MP Example</name>
 
     <properties>
-        <helidon.version>2.0.0-SNAPSHOT</helidon.version>
+        <helidon.version>2.0.0-M3</helidon.version>
```
Comment (tomas-langer) [inline examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml:29]: Same as above - this should reference snapshot

File: examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml (hunk 1 above)
Comment (ksurendra) [reply examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml:25]: Sure. I cannot build the project with "-SNAPSHOT"

File: examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml (hunk 2 above)
Comment (ksurendra) [reply examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml:29]: Sure. I cannot build the project with "-SNAPSHOT"

File: examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml (hunk 1 above)
Comment (romain-grecourt) [reply examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml:25]: Unresolving, this must reference the SNAPSHOT version. You need to build the project in order to use SNAPSHOT versions.

File: examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml (hunk 2 above)
Comment (romain-grecourt) [reply examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml:29]: Unresolving, this must reference the SNAPSHOT version. You need to build the project in order to use SNAPSHOT versions.

**Comment by ksurendra** [comment]
Re: ci/oca-check
I can see my name listed here https://www.oracle.com/technetwork/community/oca-486395.html

File: examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml (hunk 2 above)
Comment (ksurendra) [reply examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml]: > Unresolving, this must reference the SNAPSHOT version. You need to build the whole project in order to use SNAPSHOT versions.

Updated the version to "2.0.0-SNAPSHOT" https://github.com/oracle/helidon/pull/1849/commits/89cd9c4a681d5d9a48d3e0ab07593d4a5154d774 (but this maybe an oracle internal only available version?)

Comment (romain-grecourt) [reply examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml]: It's not an internal version, this is the latest version that you have to build from source.

Comment (ksurendra) [reply examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml]: Was able to successfully build the whole project using `2.0.0-SNAPSHOT`

**Review by tomas-langer — APPROVED** [review]
LGTM

**Comment by ksurendra** [comment]
@barchetta can you review this?

**Review by barchetta — APPROVED** [review]

**Pull request merged**